docs/design/pod-scheduling-readiness.md

```diff
@@ -0,0 +1,29 @@
+![Transition between states of PodGroup](./images/pod-scheduling-readiness.png)
+
+## Feature Interactions
+1. **API:** The state of a PodGroup, `PodGroupPhase` is defined in a separate api repo. Before making changes to the main repo, we need to first modify the api repo.
+```
+const (
+//...
+
+// PodGroupSchGated means the spec.SchedulingGates of at least one of the Pods of the job is not empty.
+// PodGroupSchGated is transitioned from Inqueue and back to Inqueue once spec.SchedulingGates is empty.
+PodGroupSchGated PodGroupPhase = "SchGated"
+
+//...
+)
+
+```
+2. **Plugins**: Scheduler Plugins such as proportion  will register functions to determine whether a PodGroup is enqueueable or allocatable. These functions calculate the resources already used in the cluster based on states of each PodGroup. For example, the proportion plugin determines whether a task is enqueuable when `sum(Inqueue)+sum(Running)+cur_job<TotalResource+elastic`. Since the scheduling gated jobs do not occupy resources in the cluster, by having the scheduling gated job in `SchGated` rather than `Inqueue`, they will not be summed up when calculating total used resources, which reflects the actual situation.
+3. **Actions**: Transition to `SchGated` happens to and from `Inqueue` first in the allocation action. Then, other actions will skip the gated pods.
+4. **Controllers**: One alternative design of state transition is to directly transition from `Pending` to `SchGated`. However, this is not possible because Controllers only create Pods for a job once it is inqueued (see [delayed-pod-creation](./delay-pod-creation.md)). We can only know a Pod is scheduling gated in the Inqueue state. Therefore, we need to transition to SchGated from Inqueue.
+## Granularity of Pod Scheduling Readiness
+Pod scheduling readiness is a field of the spec of a pod. However, Volcano schedules Jobs, which consists of many tasks, each corresponding to a Pod. It is possible that some of these pods are scheduling gated while others are not. To align the granularity of the Pod Scheduling Readiness and Job, we see scheduling gates as a property of a Job: as long as there is one gated pod, the job is scheduling gated. This is consistent with the workload of Volcano: most Volcano Jobs need to run as a whole and cannot be partially run.
```

Comment:
Maybe we can add that only voclano jobs have this limitation, and for normal pod/Deployment/Statefulset, etc, it's still a pod level gated feature.